### PR TITLE
kvm_io now in python3

### PR DESCRIPTION
**Pull request by AloisKlingler**
works on debian bullseye with this modification.
removed all codestyle issues, checked with "pycodestyle" on debian bullseye, besides two "line too long"

### CODE REVIEW

File: plugins/libvirt/kvm_io [hunk 1]
```diff
@@ -11,12 +11,15 @@
 # parsed environment variables:
 # vmsuffix: part of vm name to be removed
 #
-#%# capabilities=autoconf
-#%# family=contrib
+# %# capabilities=autoconf
```
Comment (sumpfralle) [inline plugins/libvirt/kvm_io:14]: The missing space here is crucial for matching these magic tokens :)

Please use the opportunity for turning the documentation into the perlpod format (see other plugins for examples).

File: plugins/libvirt/kvm_io [hunk 2]
```diff
@@ -59,54 +63,59 @@ def clean_vm_name(vm_name):
...
 
+
 def fetch(vms):
     ''' Fetch values for a list of pids
     @param dictionary {kvm_pid: cleaned vm name}
     '''
     res = {}
     for pid in vms:
-        f = open("/proc/%s/io" % pid, "r")
+        f = open("/proc/%s/io" % pid.decode("utf-8"), "r")
```
Comment (sumpfralle) [inline plugins/libvirt/kvm_io:73]: You may shorten this to `.decode()` - but this is just a matter of style.
In addition you may like to switch to `with open(...) as f:`?

(feel free to ignore these suggestions)

File: plugins/libvirt/kvm_io [hunk 3]
```diff
@@ -59,54 +63,59 @@ def clean_vm_name(vm_name):
...
+
 if __name__ == "__main__":
     if len(sys.argv) > 1:
         if sys.argv[1] in ['autoconf', 'detect']:
             if detect_kvm():
-                print "yes"
+                print("yes")
             else:
-                print "no"
+                print("no")
```
Comment (sumpfralle) [inline plugins/libvirt/kvm_io:118]: Please add a reason for the failure in brackets (e.g. `no (kvm executable missing)`).

**Review by sumpfralle — COMMENTED** [review]
Thank you for taking the time to update this plugin!

Please take a look at my comments below.

Did you check style compliance with `flake8`?

File: plugins/libvirt/kvm_io [hunk 4]
```diff
@@ -1,24 +1,49 @@
...
+
+import re
+import os
+import sys
 from subprocess import Popen, PIPE
 
+
 def config(vm_names):
-    ''' Print the plugin's config
+    ''' print(the plugin's config
```
Comment (sumpfralle) [inline plugins/libvirt/kvm_io:46]: this one here seems to be a comment - not a command :)

File: plugins/libvirt/kvm_io [hunk 5]
```diff
@@ -59,22 +85,22 @@ def clean_vm_name(vm_name):
...
-        f.close()
+        with open("/proc/%s/io" % pid.decode(), "r") as f:
+            for line in f.readlines():
+                if "read_bytes" in line:
+                    read = line.split()[1]
+                    print("%s_read.value %s" % (vms[pid], read))
+                if "write_bytes" in line:
+                    write = line.split()[1]
+                    print("%s_write.value %s" % (vms[pid], write))
+                    break
```
Comment (sumpfralle) [inline plugins/libvirt/kvm_io:102]: Just for my curiosity: why should the processing stop after `write_bytes`?
Does this imply some specific sorting of the output? Maybe the original code included this erroneously?

**Review by sumpfralle — COMMENTED** [review]
Thank you for your update!

Now I noticed two more nitpicks, which I overlooked before - sorry!

Afterwards it should be ready to be merged. Thanks!

**Comment by sumpfralle** [comment]
btw: feel free to squash and force-push your commits. Otherwise I will do it, before merging ...

File: plugins/libvirt/kvm_io (hunk 4 above)
Comment (AloisKlingler) [reply plugins/libvirt/kvm_io:46]: oh ... %s did it, not me :D

File: plugins/libvirt/kvm_io (hunk 5 above)
Comment (AloisKlingler) [reply plugins/libvirt/kvm_io:102]: it looks like it is needed.
the output in my case without the break would be not valid:
./kvm_io
KVM_NGFW_AK_read.value 7456280576
KVM_NGFW_AK_write.value 2008285184
KVM_NGFW_AK_write.value 0

whereas with the break it looks as wanted:
./kvm_io
KVM_NGFW_AK_read.value 7456280576
KVM_NGFW_AK_write.value 2008289280

Comment (sumpfralle) [reply plugins/libvirt/kvm_io:101]: OK.
In order to avoid the same question to be researched again in the future, it would be perfect to add a comment, explaining this (based on the nature of the input data stream).
But anyway: since you planned to just do the python3 migration, it is fine to leave this old stain here ...

**Comment by sumpfralle** [comment]
>> squash and force-push your commits. Otherwise I will do it, before merging ...
> sorry, I do not know what you mean with this. so please go along :-)

*squash* refers to the act of combining multiple commits into one. This can be nice for history beatification.
(other contributors are not interested in the *private* back-and-forth of a discussion/proposal)

In your case you would run `git rebase -i HEAD~~` and then select the second commit to be "squashed" into the first one.
If you never used `git rebase -i` before: go for it now - it is a great magic tool for beautification.

Anyway: I will do this for you now. But you have your private branch at for trying it on your own :)

**Comment by sumpfralle** [comment]
Thank you for your contribution!
